### DISCUSSION TITLE
Best practice if I want users to define new fields in their contacts section

### DISCUSSION THREAD

**Discussion by malalecherocks** [supabase/supabase#16412] [Questions]
Hi People 👋!

I'm creating a kind of a phone book, and my web app will be used by few companies, each of them should have the possibility to upload their customer lists, and at the beginning I added the most common fields in the table contacts such as (name, last_name, phone_number...) but I see other companies have the possibility to provide the customer the possibility to add themselves new fields of the type the want (text, number)... and in the future in the web app they can filter or search using those custom fields...

Are this companies using postgreSQL o a relational database? If I use Firestore 🔥 for example, my problem is that doing queries afterwards is a big pain... so I wonder if you can give me some best practice or ideas... I though two options using Supabase:

1️⃣ Allow each of my customers have their own "contacts" table, so they can create new columns... and I'm in charge to index it....

2️⃣ Only one single contacts table for all the customers but with an extra field of a jsonb that allows to search, but what if I have 10M rows.... ? how is going to behave?

Thanks!

**Comment by GaryAustin1** [comment]
https://medium.com/geekculture/postgres-jsonb-usage-and-performance-analysis-cdbd1242a018
Might be a good read.  There are lots of articles on jsonb performance with Postgres out there.
JSONB seems like the best approach for what you want at least from a management point of view.
Also would 10M rows be for one company or many?  As you filter other columns, the jsonb filter would have less rows to process.

**Reply by malalecherocks** [reply]
thanks a lot, I knew you were going to reply... thanks a lot! I'm going to read about it....
